core/db_classes/EE_Line_Item.class.php

```diff
@@ -1554,6 +1554,9 @@ public function taxable_total()
                     }
                 } elseif ($child_line_item->type() === EEM_Line_Item::type_sub_total) {
                     $total += $child_line_item->taxable_total();
+                } elseif ($child_line_item->is_percent()) {
+                    // Taxes should be charged on the discounted price.
+                    $total += $child_line_item->total();
```

Comment:
hmmm... this doesn't feel right to me...
why is the line item in question not being caught by the first clause above?

ie:
```php 
if ($child_line_item->type() === EEM_Line_Item::type_line_item && $child_line_item->is_taxable()) {
```

which already has some logic for handling precentages.

This new clause looks like it is going to add all percentage sub line items to the running total, which doesn't seem correct.

Can you plz show me an example of the kind of line item that would make it into this clause?

Reply:
>why is the line item in question not being caught by the first clause above?

Because a discounts is `! is_taxable()`. But now I think I know why, after you pointed out that promotions setting...

> This new clause looks like it is going to add all percentage sub line items to the running total

Aren't all the line items of type `"percentage"` considered to only be promotions ? Tax amounts in percent aren't counted as `true` in `is_percent()`. What else could be `true` in `is_percent()` beside a promotion ? Maybe I'm missing some. But a few PMs do use `is_percent()` to filter out only promos.

Anyways, this change is not suitable here, I totally agree. Your doubts did help figure this out. Thank you.